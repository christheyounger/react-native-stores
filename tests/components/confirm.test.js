import React from 'react';
import ConfirmModal from '../../app/components/confirm';
import renderer from 'react-test-renderer';
import { Modal, Text, View, Button } from 'react-native';

describe('component.ConfirmModal', function() {
  const handleClick = jest.fn().mockResolvedValue({})
  const store = {id: 12, name: 'Root Store', branches: [{id: 14, name: 'Child Store', branches: []}]};
  const wrapper = renderer.create(
    <ConfirmModal
      color='red'
      label='Delete'
      title={`Delete Something`}
      handleClick={handleClick}
    />
  );
  test('renders correctly', function() {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  test('can open modal', () => {
    expect(wrapper.root.findByType(Modal).props.visible).toEqual(false);
    wrapper.root.findAllByType(Button)[0].props.onPress();
    expect(wrapper.root.findByType(Modal).props.visible).toEqual(true);
  });
  test('can close modal', () => {
    wrapper.root.findAllByType(Button)[0].props.onPress();
    expect(wrapper.root.findByType(Modal).props.visible).toEqual(true);
    wrapper.root.findByType(Modal).props.onRequestClose();
    expect(wrapper.root.findByType(Modal).props.visible).toEqual(false);
  });
  test('can cancel modal', () => {
    wrapper.root.findAllByType(Button)[0].props.onPress();
    expect(wrapper.root.findByType(Modal).props.visible).toEqual(true);
    wrapper.root.findByType(Modal).findByProps({title: "No, don't"}).props.onPress();
    expect(wrapper.root.findByType(Modal).props.visible).toEqual(false);
  });
  test('can submit modal', async () => {
    wrapper.root.findAllByType(Button)[0].props.onPress();
    expect(wrapper.root.findByType(Modal).props.visible).toEqual(true);
    await wrapper.root.findByType(Modal).findByProps({title: "Delete"}).props.onPress();
    expect(handleClick).toHaveBeenCalled();
    expect(wrapper.root.findByType(Modal).props.visible).toEqual(false);
  });
});
