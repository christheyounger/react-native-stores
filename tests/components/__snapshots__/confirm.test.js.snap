// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component.ConfirmModal renders correctly 1`] = `
<View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "red",
            },
          ]
        }
      >
        Delete
      </Text>
    </View>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "stretch",
          "flexDirection": "column",
          "marginTop": 22,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "stretch",
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 24,
            }
          }
        >
          Delete Something
        </Text>
        <Text
          style={
            Object {
              "padding": 20,
            }
          }
        >
          Are you sure you wish to 
          Delete Something
          ?
        </Text>
        <View
          style={
            Object {
              "alignItems": "stretch",
              "flexDirection": "column",
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityStates={Array []}
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "padding": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                No, don't
              </Text>
            </View>
          </View>
          <View
            accessibilityRole="button"
            accessibilityStates={Array []}
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "padding": 8,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "red",
                    },
                  ]
                }
              >
                Delete
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
